server/index.js

```diff
@@ -54,6 +54,100 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!req.session.cartId) {
+    return res.json([]);
+  } else {
+    const sql = `
+  select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+  from "cartItems" as "c"
+  join "products" as "p" using ("productId")
+ where "c"."cartId" = $1
+  `;
+    const params = [cartId];
+    db.query(sql, params)
+      .then(result => res.json(result.rows));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = Number(req.body.productId);
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return res.status(400).json({ error: '"productId" must be a postive integer' });
+  }
+  const sql = `
+  select "price"
+  from "products"
+  where "productId" = $1
+`;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      const price = product.price;
+
+      if (!price) {
+        throw (new ClientError('"productId" cannot be found', 400));
+      } else {
+        const sql = `
+        insert into "carts" ("cartId", "createdAt")
+        values (default, default)
+        returning "cartId"
+        `;
+        return db.query(sql)
+          .then(result => {
+            const cartId = result.rows[0];
+            return {
+              cartId: cartId.cartId,
+              price: price
+            };
+          });
+      }
```

Comment:
There is a step in the instructions where you need to add some logic here. If there is already a `cartId` on `req.session`, then avoid creating a new cart and just reuse the `req.session.cartId`. You'll know this works when you can `POST` multiple items to the cart and then issue a `GET` to the cart and get _all_ of those items.